R/calculate_tdm.R

```diff
@@ -188,23 +190,37 @@ add_tdm <- function(data, groups) {
   data %>%
     group_by(!!!rlang::syms(groups)) %>%
     mutate(
-      .numerator = .data$scenario_production_plus_delta_t2 - .data$portfolio_production_plus_delta_t1,
+      .numerator_portfolio = .data$scenario_production_plus_delta_t2 - .data$portfolio_production_plus_delta_t1,
+      .numerator_scenario = .data$scenario_production_plus_delta_t2 - .data$scenario_production_plus_delta_t1,
       .denominator = .data$scenario_production_plus_delta_t2 - .data$scenario_production_t0,
-      # TODO: This case we
-      tdm_technology_value = ifelse(
+      tdm_technology_value_portfolio = ifelse(
         .data$.denominator == 0,
         0,
-        max(0, (.data$.numerator / .data$.denominator) * .data$monotonic_factor) * 2
+        max(0, (.data$.numerator_portfolio / .data$.denominator) * .data$monotonic_factor) * 2
       ),
-      .numerator = NULL,
+      tdm_technology_value_scenario = ifelse(
+        .data$.denominator == 0,
+        0,
+        max(0, (.data$.numerator_scenario / .data$.denominator) * .data$monotonic_factor) * 2
+      ),
+      .numerator_portfolio = NULL,
+      .numerator_scenario = NULL,
```

Comment:
This just doubles up the calculation in a DRY-ish way. `denominator` is the same for both, only the `numerator` changes between `portfolio` and `scenario`.